http_handlers.go

```diff
@@ -280,6 +280,94 @@ func handleLearn(c echo.Context) error {
 	return c.JSON(http.StatusOK, "success")
 }
 
+func handleLearnFileUpload(c echo.Context) error {
+	var (
+		app = c.Get("app").(*App)
+	)
+
+	// Multipart form
+	form, err := c.MultipartForm()
+	if err != nil {
+		return err
+	}
+
+	langCode := c.FormValue("lang")
+	files := form.File["files"]
+
+	_, ok := trainChannel[langCode]
+	if !ok {
+		app.log.Printf("unknown language requested to learn: %s", langCode)
+		return echo.NewHTTPError(http.StatusBadRequest, "unable to find language to train")
+	}
+
+	if len(files) == 0 {
+		return echo.NewHTTPError(http.StatusBadRequest, "No file uploaded")
+	}
+
+	// Copy files first
+	for _, file := range files {
+		// Source
+		src, err := file.Open()
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+		}
+		defer src.Close()
+
+		learnFilePath := getSyncMetadataDir() + "/learn-" + langCode + "-" + file.Filename
+
+		// Destination
+		dst, err := os.Create(learnFilePath)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+		}
+		defer dst.Close()
+
+		// Copy
+		if _, err = io.Copy(dst, src); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+		}
+	}
+
+	learnFromFile := func(langCode, fileToLearn string) {
```

Comment:
it seems like a bad practice. Refactor into a separate function.

Reply:
@joicemjoseph Thought of it as a callbacks like in JavaScript, it's not preferred in Go eh ?